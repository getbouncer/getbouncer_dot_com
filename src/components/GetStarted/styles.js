import styled from '@emotion/styled'

export const Icon = styled.img`
  width: 72px;
`

export const Headline = styled.h2`
  font-size: 2.857rem;
  margin: 14px 0 24px;
`

export const Text = styled.div`
  margin: 0 auto 30px;
  max-width: 476px;
`
