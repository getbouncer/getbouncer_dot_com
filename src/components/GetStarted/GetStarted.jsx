import React from 'react'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'
import { css } from '@emotion/core'
import CtaButton from '~components/CtaButton'
import { breakpoints } from '~styles'
import { Icon, Headline, Text } from './styles'

export const PureGetStarted = ({ data }) => {
  const { background, cta_text: ctaText, icon, text, title } = data
  const backgroundStyles = css`
    background-image: url(${background.url});
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
    color: #fff;
    padding: 120px 0;
    text-align: center;

    @media ${breakpoints.smartphone} {
      padding: 60px 32px 68px;
    }
  `

  return (
    <section css={backgroundStyles}>
      <Icon src={icon.url} alt={icon.alt} />
      <Headline>{title.text}</Headline>
      <Text dangerouslySetInnerHTML={{ __html: text.html }} />
      <CtaButton isInternal text={ctaText} to="/contact" inverted />
    </section>
  )
}

PureGetStarted.propTypes = {
  data: PropTypes.shape().isRequired,
}

export const GetStarted = () => {
  const data = useStaticQuery(graphql`
    query GetStartedQuery {
      prismicGlobals {
        data {
          background {
            url
          }
          cta_text
          icon {
            url
            alt
          }
          title {
            text
          }
          text {
            html
          }
        }
      }
    }
  `)

  return <PureGetStarted data={data.prismicGlobals.data} />
}

export default GetStarted
