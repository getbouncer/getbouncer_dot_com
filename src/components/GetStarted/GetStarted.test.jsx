import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import { PureGetStarted as GetStarted } from './GetStarted'

describe('GetStarted', () => {
  it('renders correctly', () => {
    const data = {
      background: {
        url: 'www.example.com/background.png',
      },
      cta_text: 'cta text',
      icon: {
        url: 'www.example.com/icon.svg',
        alt: 'icon',
      },
      title: {
        html: 'title',
      },
      text: {
        html: '<p>text</p>',
      },
    }
    const tree = renderer
      .create(<GetStarted data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
