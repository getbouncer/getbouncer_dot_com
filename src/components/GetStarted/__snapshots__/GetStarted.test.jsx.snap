// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "12bwd2y",
      "next": undefined,
      "styles": "
    background-image: url(www.example.com/background.png);
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
    color: #fff;
    padding: 120px 0;
    text-align: center;

    @media (max-width: 920px) {
      padding: 60px 32px 68px;
    }
  ",
      "toString": [Function],
    }
  }
>
  <img
    alt="icon"
    className="css-1y3nu44"
    src="www.example.com/icon.svg"
  />
  <h2
    className="css-7x7zvy"
  />
  <div
    className="css-8xi3kf"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>text</p>",
      }
    }
  />
  <a
    className="css-1acz3hk"
    href="/contact"
    inverted={true}
  >
    cta text
  </a>
</section>
`;
